frontends/web/src/routes/account/summary/summarybalance.tsx

```diff
@@ -55,73 +63,87 @@ export function SummaryBalance ({
+        {connected ?
+          <Badge
+            icon={props => <USBSuccess {...props} />}
+            type="success"
+          >
+            {t('device.keystoreConnected')}
+          </Badge> :
+          null
+        }
+      </div>
+      <div className={style.balanceTable}>
+        <table className={style.table}>
+          <colgroup>
+            <col width="33%" />
+            <col width="33%" />
+            <col width="*" />
+          </colgroup>
+          <thead>
+            <tr>
+              <th>{t('accountSummary.name')}</th>
+              <th>{t('accountSummary.balance')}</th>
+              <th>{t('accountSummary.fiatBalance')}</th>
+            </tr>
+          </thead>
+          <tbody>
+            { accounts.length > 0 ? (
+              coins.map(coinCode => {
+                const balanceRows = accountsPerCoin[coinCode].map(account =>
+                  <BalanceRow
+                    key={account.code}
+                    code={account.code}
+                    name={account.name}
                     coinCode={account.coinCode}
-                    coinName={account.coinName}
-                    balance={totalBalancePerCoin && totalBalancePerCoin[coinCode]}
-                  />);
-              }
-              return balanceRows;
-            })
-          ) : (
+                    balance={balances && balances[account.code]}
+                  />
+                );
+                if (balanceRows?.length > 1) {
+                  const account = accountsPerCoin[coinCode][0];
+                  balanceRows.push(
+                    <SubTotalRow
+                      key={account.coinCode}
+                      coinCode={account.coinCode}
+                      coinName={account.coinName}
+                      balance={totalBalancePerCoin && totalBalancePerCoin[coinCode]}
+                    />);
+                }
+                return balanceRows;
+              })
+            ) : (
+              <tr>
+                <td colSpan={3} className={style.noAccount}>
+                  {t('accountSummary.noAccount')}
+                </td>
+              </tr>
+            )}
+          </tbody>
+          <tfoot>
             <tr>
-              <td colSpan={3} className={style.noAccount}>
-                {t('accountSummary.noAccount')}
+              <th>
+                <strong>{t('accountSummary.total')}</strong>
+              </th>
+              <td colSpan={2}>
+                {totalBalance ? (
+                  <>
+                    <strong>
+                      <Amount amount={totalBalance.total} unit={totalBalance.fiatUnit}/>
+                    </strong>
+                    {' '}
+                    <span className={style.coinUnit}>
+                      {totalBalance.fiatUnit}
+                    </span>
+                  </>
+                ) : (<Skeleton />) }
               </td>
```

Comment:
**nit:** this has always been there, but I generally find this component to be a bit hard to read due to the number of conditional renders.

I think we could make the component easier to read by extracting these components into its own. E.g, here we could have `<TotalBalance ... />` component - also applies to the one above (lets say `TotalCoinBalance` or smt).

Just a nit, we could also do it another time ofc :)

Reply:
Good point. Done! I only did it for the TotalBalance, as the coin total looks already good to me with the `SubTotalRow` component. Did you have something different in mind?